### PR TITLE
ENH: Use date and datetime objects

### LINKED ISSUE #11327: info['subject_info']['birthday'] should support a datetime object
### Describe the new feature or enhancement

Working on https://mne.discourse.group/t/minor-priority-issue-with-converting-mne-raw-into-bids-with-mne-bids-age-birthday-info-of-subject/5922, I realized that a subject's birthday must be specified as a tuple of ints. I believe this is inconsistent with the way we allow to specify the measurement date (datetime object).

### Describe your proposed implementation

It should be possible to specify `Info['subject_info']['birthday']` as a `datetime.datetime` object. 

### Describe possible alternatives

–

### Additional context

_No response_

### LINKED ISSUE #11966: BUG: `raw.anonymize()` not fully anonymizing
### Description of the problem

When anonymizing a `raw` object with func `anonymize()`, the subject info and `meas_date` are still accessible through `raw._raw_extras` attribute as shown in the below example.

Regarding subject info, it would be easier in my opinion to remove field "subject_info" from both `raw.info` and `*raw._raw_extras` (or simply empty unecessary fields, keeping "his_id" only for instance) as it is not expected to be able to retrieve the information afterwards by definition.

### Steps to reproduce

```Python
import mne

data_path = mne.datasets.testing.data_path(download=False)
fpath = data_path / "EDF" / "test_edf_stim_resamp.edf"

raw = mne.io.read_raw_edf(fpath, verbose=False)

print("Subject info before anonymization")
print("\t", raw.info["subject_info"])
print("\t", raw.info["meas_date"])
print("\t", raw._raw_extras[0]["subject_info"])
print("\t", raw._raw_extras[0]["meas_date"])

raw.anonymize()

print("\t", "\nSubject info after anonymization")
print("\t", raw.info["subject_info"])
print("\t", raw.info["meas_date"])
print("\t", raw._raw_extras[0]["subject_info"])
print("\t", raw._raw_extras[0]["meas_date"])
print(raw._raw_extras[0]["subject_info"])
```

### Expected results

```
Subject info before anonymization
         {'his_id': 'X', 'sex': 1, 'last_name': 'X', 'birthday': (1967, 10, 9)}
         2016-02-14 10:33:35+00:00
         {'id': 'X', 'sex': 'M', 'birthday': datetime.datetime(1967, 10, 9, 0, 0), 'name': 'X'}
         2016-02-14 10:33:35+00:00

Subject info after anonymization
         {'his_id': '0', 'sex': 0, 'last_name': 'mne_anonymize', 'birthday': (1951, 8, 25)}
         2000-01-01 00:00:00+00:00
         {'id': '0', 'sex': 'unknown', 'birthday': datetime.datetime(1951, 8, 25, 0, 0), 'name': 'mne_anonymize'}
         2000-01-01 00:00:00+00:00
```
or alternatively

```
Subject info before anonymization
         {'his_id': 'X', 'sex': 1, 'last_name': 'X', 'birthday': (1967, 10, 9)}
         2016-02-14 10:33:35+00:00
         {'id': 'X', 'sex': 'M', 'birthday': datetime.datetime(1967, 10, 9, 0, 0), 'name': 'X'}
         2016-02-14 10:33:35+00:00

Subject info after anonymization
         {'his_id': 'X'}
         2000-01-01 00:00:00+00:00
         {}
         2000-01-01 00:00:00+00:00
```

### Actual results

```
Subject info before anonymization
         {'his_id': 'X', 'sex': 1, 'last_name': 'X', 'birthday': (1967, 10, 9)}
         2016-02-14 10:33:35+00:00
         {'id': 'X', 'sex': 'M', 'birthday': datetime.datetime(1967, 10, 9, 0, 0), 'name': 'X'}
         2016-02-14 10:33:35+00:00

Subject info after anonymization
         {'his_id': '0', 'sex': 0, 'last_name': 'mne_anonymize', 'birthday': (1951, 8, 25)}
         2000-01-01 00:00:00+00:00
         {'id': 'X', 'sex': 'M', 'birthday': datetime.datetime(1967, 10, 9, 0, 0), 'name': 'X'}
         2016-02-14 10:33:35+00:00
```

### Additional information

The above code uses the latest release (`mne==1.5.1`).

### PR DESCRIPTION
**Pull request by larsoner**
1. Use `datetime.date` for `info["subject_info"]["birthday"]` rather than a year/month/day tuple
2. Use `datetime.datetime` for `info["helium_info"]["meas_date"]`
3. Fix in-memory anonymization for EDF by deleting `edf_info["subject_info"]` and `edf_info["meas_date"]`

Closes #11966
Closes #11327

### CODE REVIEW

File: mne/_fiff/meas_info.py [hunk 1]
```diff
@@ -1331,9 +1345,13 @@ class Info(dict, SetChannelsMixin, MontageMixin, ContainsMixin):
             Helium level (%) after position correction.
         orig_file_guid : str
             Original file GUID.
-        meas_date : tuple of int
+        meas_date : instance of datetime.datetime
             The helium level meas date.
 
+            .. versionchanged:: 1.7
```
Comment (hoechenberger) [inline mne/_fiff/meas_info.py:1351]: 1.8,  no?

File: mne/_fiff/meas_info.py [hunk 2]
```diff
@@ -1331,9 +1345,13 @@ class Info(dict, SetChannelsMixin, MontageMixin, ContainsMixin):
             Helium level (%) after position correction.
         orig_file_guid : str
             Original file GUID.
-        meas_date : tuple of int
+        meas_date : instance of datetime.datetime
```
Comment (cbrnr) [inline mne/_fiff/meas_info.py:1348]: Why not just `datetime.datetime`? We don't write e.g. `instance of int` either.

**Comment by larsoner** [comment]
Thanks for the quick reviews, should be good now!

**Review by drammock — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

**Comment by hoechenberger** [comment]
thanks Eric, very nice

**Pull request merged**